java/test/jmri/jmrit/logix/LearnWarrantTest.java

```diff
@@ -201,25 +200,31 @@ private Sensor recordtimes(String[] route, DccThrottle throttle) throws Exceptio
     @BeforeEach
     public void setUp() {
         JUnitUtil.setUp();
+        JUnitUtil.resetInstanceManager();
         jmri.util.JUnitUtil.resetProfileManager();
         JUnitUtil.initConfigureManager();
         JUnitUtil.initInternalTurnoutManager();
-        JUnitUtil.initInternalLightManager();
         JUnitUtil.initInternalSensorManager();
-        JUnitUtil.initInternalSignalHeadManager();
         JUnitUtil.initDebugPowerManager();
-        JUnitUtil.initDebugThrottleManager();
-        JUnitUtil.initMemoryManager();
         JUnitUtil.initOBlockManager();
         JUnitUtil.initLogixManager();
-        JUnitUtil.initConditionalManager();
         WarrantPreferences.getDefault().setShutdown(WarrantPreferences.Shutdown.NO_MERGE);
         JUnitUtil.initWarrantManager();
+        JUnitUtil.initDebugThrottleManager();
     }
 
     @AfterEach
     public void tearDown() {
-        JUnitUtil.clearShutDownManager(); // should be converted to check of scheduled ShutDownActions
+        if (InstanceManager.containsDefault(ShutDownManager.class)) {
+            ShutDownManager sm = InstanceManager.getDefault(jmri.ShutDownManager.class);
+            List<Runnable> rlist = sm.getRunnables();
+            while (!rlist.isEmpty()) {
+                if (rlist.get(0) instanceof jmri.jmrit.logix.WarrantShutdownTask) {
+                    sm.deregister((ShutDownTask)rlist.get(0));
+                    rlist = sm.getRunnables();
+                }
+            }
```

Comment:
If `rlist.get(0)` is not a `WarrantShutdownTask`, this will be an infinite loop. I suggest an `else` statement that throws an exception or breaks out of the loop.